frontend/src/pages/UserPage.jsx

```diff
@@ -1,13 +1,178 @@
-import { useParams } from 'react-router';
+import React, { useEffect, useState, useRef } from 'react';
+import { useParams, useNavigate, Link } from 'react-router';
+import { useUser } from '../hooks/useUser';
+import { useApi } from '../hooks/useApi';
+import Spinny from '../components/Spinny';
+
 const UserPage = () => {
-  const { id } = useParams(); // Get the user ID from the URL
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { isLogged, isAdmin } = useUser();
+  const [userData, setUserData] = useState(null);
+  const [showDeactivateModal, setShowDeactivateModal] = useState(false);
+  const deactivateModalRef = useRef(null);
+
+  const {
+    data: fetchUserData,
+    error: fetchUserError,
+    isLoading: isUserLoading,
+    fetchFromApi: fetchUser,
+  } = useApi({ method: 'GET' });
+
+  const {
+    data: deactivateData,
+    error: deactivateError,
+    isLoading: isDeactivating,
+    fetchFromApi: deactivateUser,
+  } = useApi({ method: 'DELETE' });
+
+  useEffect(() => {
+    if (!isLogged) {
+      navigate('/login');
+      return;
+    }
+    if (!isAdmin) {
+      navigate('/');
+      return;
+    }
+
+    if (id) {
+      fetchUser(`/users/${id}`);
+    }
+  }, [id, isLogged, isAdmin, navigate, fetchUser]);
+
+  useEffect(() => {
+    if (fetchUserData && !fetchUserError) {
+      setUserData(fetchUserData.user);
```

Comment:
```suggestion
      setUserData(fetchUserData);
```
Fetched data doesn't have `user` attribute.